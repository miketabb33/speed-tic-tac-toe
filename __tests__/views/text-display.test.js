import React from 'react'
import renderer from 'react-test-renderer'
import TextDisplay from '../../views/text-display.js'

test('render xFirst', () => {
	const subject = renderer.create(
		<TextDisplay value = { TextDisplay.xFirst } />
	)

	const result = subject.toJSON()
	expect(result).toMatchSnapshot()
})

test('render xTurn', () => {
	const subject = renderer.create(
		<TextDisplay value = { TextDisplay.turn("X") } />
	)

	const result = subject.toJSON()
	expect(result).toMatchSnapshot()
})

test('render oTurn', () => {
	const subject = renderer.create(
		<TextDisplay value = { TextDisplay.turn("O") } />
	)

	const result = subject.toJSON()
	expect(result).toMatchSnapshot()
})

test('render xWin', () => {
	const subject = renderer.create(
		<TextDisplay value = { TextDisplay.winner("X") } />
	)

	const result = subject.toJSON()
	expect(result).toMatchSnapshot()
})

test('render oWin', () => {
	const subject = renderer.create(
		<TextDisplay value = { TextDisplay.winner("O") } />
	)

	const result = subject.toJSON()
	expect(result).toMatchSnapshot()
})

test('render draw', () => {
	const subject = renderer.create(
		<TextDisplay value = { TextDisplay.draw } />
	)

	const result = subject.toJSON()
	expect(result).toMatchSnapshot()
})