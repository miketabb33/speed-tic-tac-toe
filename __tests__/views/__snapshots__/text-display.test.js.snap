// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render draw 1`] = `
<div
  className="textDisplay"
>
  Draw
</div>
`;

exports[`render oTurn 1`] = `
<div
  className="textDisplay"
>
  O's Turn
</div>
`;

exports[`render oWin 1`] = `
<div
  className="textDisplay"
>
  O is the Winner!
</div>
`;

exports[`render xFirst 1`] = `
<div
  className="textDisplay"
>
  X Goes First
</div>
`;

exports[`render xTurn 1`] = `
<div
  className="textDisplay"
>
  X's Turn
</div>
`;

exports[`render xWin 1`] = `
<div
  className="textDisplay"
>
  X is the Winner!
</div>
`;
